packages/desktop-gui/src/app/ui-blocker.jsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import { createPortal } from 'react-dom'
+import { observer } from 'mobx-react'
+
+import appStore from '../lib/app-store'
+
+const UiBlocker = observer(() => {
+  if (!appStore.isUiBlocked) return null
+
+  return createPortal(
+    <div className='ui-blocker' />,
+    document.getElementById('ui-blocker')
+  )
+})
+
+export default UiBlocker
```

Comment:
Can we change the name of this to be Scrim. While ui-blocker is descriptive, I think Scrim is a more widely accepted name for what this component represents. For example, here is a commonly accepted definition from Google's [material design of a scrim](https://material.io/design/environment/surfaces.html#attributes) (scroll down to scrim; I could not link directly to the section)